### DISCUSSION TITLE
How to match against array query params using `intercept`?

### DISCUSSION THREAD

**Discussion by chipit24** [cypress-io/cypress#26404] [Questions and Help] [answered]
I have a request going out that looks like this:
```
/api/transactions?per_page=25&page=1&ids[]=1
```

I set up my interceptor in Cypress like so:

```
cy.intercept(
  {
    method: "GET",
    pathname: "/api/transactions",
    query: {
      per_page: 25,
      page: 1,
      "ids[]": "1",
    },
  },
  { body: { data: "some data" } },
);
```

I tried `"ids[1]": "1"` and `"ids": "1"` but nothing seems to work to match against "array" query strings. Anyone else run into this before?

Cypress seems to just ignore the `"ids"` query param; I have no problem matching on the "single string" query params.

FYI: This request is being sent by [`axios`](https://github.com/axios/axios), which unwraps array-like param objects as separate fields with the same key, hence why it's passing the query along as `ids[]=1&ids[]=2`. Not sure how Cypress supports matching against this even if only one item is sent?

**Comment by ntwind** [comment]
as it turns, man,  the only way for now is using reg exp, hope your params stay always in same order

**Comment by jennifer-shehane** [comment, marked as answer]
The query matcher compares string values, so repeated/array params (?ids=1&ids=2) need a RegExp on the URL or a function matcher:

```js
cy.intercept({ method: 'GET', url: /\/items\?(.*&)?ids=/ }).as('items')
// or inspect inside a middleware/handler:
cy.intercept('/items*', (req) => {
  const ids = new URL(req.url).searchParams.getAll('ids')
  if (ids.includes('1')) req.reply({ fixture: 'items.json' })
})
```
